cmd/shp/main.go

```diff
@@ -51,4 +67,10 @@ func initPFlags() {
 	flags := pflag.NewFlagSet(ApplicationName, pflag.ExitOnError)
 	flags.AddGoFlagSet(goflag.CommandLine)
 	pflag.CommandLine = flags
+
+	for _, flag := range hiddenLogFlags {
+		if err := flags.MarkHidden(flag); err != nil {
```

Comment:
Interesting approach. Early on I was trying to achieve the same but using [ConfigOverrideFlags](https://github.com/kubernetes/client-go/blob/af4295f5013b55c45b05f1e7850f7fa9c4dbfdb4/tools/clientcmd/overrides.go#L40-L48), and it does take more effort to understand and adapt the existing bits.